### ISSUE TITLE
Spring 2.0.0.M1 springfox  getPluginFor method did not exist

### ISSUE THREAD

**Issue by jkeruzec** [springfox/springfox#2968]
I'm trying to use Springfox with latest version of Spring. 
I think there is some issue with not found method :

> ***************************
> APPLICATION FAILED TO START
> ***************************
> 
> Description:
> 
> An attempt was made to call a method that does not exist. The attempt was made from the following location:
> 
>     springfox.documentation.spring.web.plugins.DocumentationPluginsManager.createContextBuilder(DocumentationPluginsManager.java:152)
> 
> The following method did not exist:
> 
>     org.springframework.plugin.core.PluginRegistry.getPluginFor(Ljava/lang/Object;Lorg/springframework/plugin/core/Plugin;)Lorg/springframework/plugin/core/Plugin;
> 
> The method's class, org.springframework.plugin.core.PluginRegistry, is available from the following locations:
> 
>     jar:file:/C:/Users/Julien.Keruzec/.m2/repository/org/springframework/plugin/spring-plugin-core/2.0.0.M1/spring-plugin-core-2.0.0.M1.jar!/org/springframework/plugin/core/PluginRegistry.class
> 
> It was loaded from the following location:
> 
>     file:/C:/Users/Julien.Keruzec/.m2/repository/org/springframework/plugin/spring-plugin-core/2.0.0.M1/spring-plugin-core-2.0.0.M1.jar
> 
> 
> Action:
> 
> Correct the classpath of your application so that it contains a single, compatible version of org.springframework.plugin.core.PluginRegistry

**Comment by jaumzors** [comment]
Same thing here, cannot start the app using swagger 2 and spring boot 2.2.0.M2

**Comment by ebussieres** [comment]
Duplicate of https://github.com/springfox/springfox/issues/2932

**Comment by modikanisha** [comment]
Hey @jaumzors ,
Have you found any solution for this issue. I have faced same issue. Please help me.

> Same thing here, cannot start the app using swagger 2 and spring boot 2.2.0.M2

**Comment by jkeruzec** [comment]
@modikanisha 

```
<!-- Work around springfox -->
<dependency>
	<groupId>org.springframework.plugin</groupId>
	<artifactId>spring-plugin-core</artifactId>
	<version>1.2.0.RELEASE</version>
</dependency>
```

**Comment by hendisantika** [comment]
I just add this:
```
<!-- https://mvnrepository.com/artifact/org.springframework.plugin/spring-plugin-core -->
        <dependency>
            <groupId>org.springframework.plugin</groupId>
            <artifactId>spring-plugin-core</artifactId>
            <version>2.0.0.RELEASE</version>
        </dependency>
```

**Comment by MohamedLamkhayar** [comment]
> <dependency>
>             <groupId>org.springframework.plugin</groupId>
>             <artifactId>spring-plugin-core</artifactId>
>             <version>2.0.0.RELEASE</version>
>         </dependency>

THANK you

**Comment by dilipkrish** [comment]
This has been fixed in 3.0.0
